test/session_test.dart

```diff
@@ -289,6 +291,39 @@ Future<void> main([List<String>? args]) async {
 
     realm.syncSession.raiseSessionError(SyncErrorCategory.session, 100, false);
   });
+
+  baasTest('SyncSession.getConnectionStateStream', (configuration) async {
+    final realm = await getIntegrationRealm();
+
+    await waitForCondition(() => realm.syncSession.connectionState == ConnectionState.connected);
+
+    final states = <ConnectionStateChange>[];
+    final stream = realm.syncSession.getConnectionStateStream();
```

Comment:
There are specific support for testing streams, such as `emitInOrder` etc. that can make these tests more elegant.

Reply:
I'm a noob when it comes to streams and tests, so couldn't figure out how to do it - my main issue is that I need to do something to trigger emitting the events, which I don't see an obvious way to do.